### PR TITLE
Fix generative join, nil handling in events

### LINKED ISSUE #72: The generative join chokes on missing data
In a real-world example using the SPPL backend, the following works fine,
```
    (SELECT * FROM table WHERE age IS NOT NULL)
    GENERATIVE JOIN model GIVEN age
```
but stops working when the table is not filtered for null conditions.

```
    (SELECT * FROM table)
    GENERATIVE JOIN model GIVEN age
```
:bulb: This was tested using SPPL backend. It might be OK with ClojureCat.

### PR DESCRIPTION
**Pull request by zane**
## What does this do?

- Removes a couple unused expressions that were accidentally committed.
- Upgrades the dependency on inferenceql.inference to `HEAD`.
- During a generative join when evaluating the model expression, treat values from the row as a "tuple", thus causing them to be nil-safe.
- Changes condition and constrain such that if the event contains `nil`s the `nil`s are ignored. Previously if the event contained `nils` the _entire event_ was ignored.
- Resolves #72.

## Why should we do this?

- #72
- It appears that currently conditioning and constraining on the values in a row will do nothing if any of the values in the row are `nil`. 😱 

## How do I test this?

This pull request includes some unit tests, but additional manual testing would be appreciated.

### CODE REVIEW

**Review by Schaechtle — APPROVED** [review]
# TL;DR
**Works very well addressing the issues with generative join (e.g. [here](https://github.com/InferenceQL/inferenceql.query/blob/schaechtle/repro-gen-join-issue/repro.clj)). Note: iql.query now throws errors in cases where before, it gave reasonable results but that's what we want.**

### There are new errors but overall, the behavior is now sound - before it wasn't.
This PR *does* make constraining on data in a table throw errors with SPPL that weren't triggered before - see the following example produced with the data and model used [here](https://github.com/InferenceQL/inferenceql.query/blob/schaechtle/repro-gen-join-issue/repro.clj). The query causes iql.query's main branch to ignore  the constraint but throws an error on this branch (zane/generative-join-fix):
```clojure
(permissive/query 
  "SELECT PROBABILITY OF foo UNDER model_spe GIVEN bar > data_with_missing.bar FROM data_with_missing LIMIT 1"
  database)
```
While this happened to behave reasonably on iql.query main - here is the error it throws on zane/generative-join-fix:
```
Execution error at libpython-clj2.python.ffi/check-error-throw (ffi.clj:708).
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/sppl/spe.py", line 170, in condition
    event_dnf = dnf_normalize(event)
  File "/usr/local/lib/python3.10/dist-packages/sppl/dnf.py", line 73, in dnf_normalize
    event_dnf = event.to_dnf()
AttributeError: 'NotImplementedType' object has no attribute 'to_dnf'   
```
I write *"happened to behave reasonably"* above because ignoring the constraint above is not completely insane. However,  it's easy to produce erroneous results:
```clojure
(strict/query "SELECT PROBABILITY DENSITY OF VAR foo = foo UNDER model_spe CONSTRAINED BY VAR bar > bar AND VAR bar > 42 FROM data_with_missing LIMIT 1" database)
``` 

The actual error message isn't helpful but throwing an error is what we want - unless there are clear semantics that I am not seeing.
